pkg/s3wrapper/filesystem.go

```diff
@@ -52,24 +53,73 @@ func (f *FSClient) CreateBucket() error {
 	return nil
 }
 
-func (f *FSClient) Upload(ctx context.Context, data []byte, objectName string) error {
+const xattrUserAttributePrefix = "user."
+
+func (f *FSClient) writeFileMetadata(filePath string, metadata map[string]*string) error {
+	for attributeName, attributeValue := range metadata {
+		err := xattr.Set(filePath, fmt.Sprintf("%s%s", xattrUserAttributePrefix, attributeName), []byte(*attributeValue))
```

Comment:
What if `attributeValue` is nil? If it can't be nil then why is it a pointer?

Reply:
This is a pointer because the S3 API defines it as a pointer
I don't believe it is ever nil but we have no control over how Amazon has built the API

Reply:
Driven by S3 API, in our use case, this will never be nil. As suggested by @pastequo I will make an effort to change this to `map[string]string` instead of `map[string]*string`

Then we should only need to handle this issue in a single location.

Suggest that we fail the operation if a nil is detected in this metadata field once we have done so.

Reply:
Closing as the whole topic of map[string]*string vs map[string]string is dealt with in another comment.